Dockerfile

```diff
@@ -8,14 +8,16 @@ ENV BUNDLE_FROZEN="true"
 
 WORKDIR /app
 RUN apk --update --no-cache add git build-base postgresql-dev
-
-COPY . /app
 RUN gem update bundler --no-document
+
+# NB: its more efficient not to copy the full app folder until after the gems are installed (reduces unnecessary rebuilds)
+COPY Gemfile Gemfile.lock .ruby-version /app/
```

Comment:
Thanks!  :)

Reply:
Having a bit more though of why this is.   If we are building our gems in `/usr/local/bundle` and not in  `app/vendor` would it make a difference? 

I did test both cases locally and couldn't see a difference.
(both starting from a fresh all-images-deleted docker engine) 
```
build -t master .                       3.06s user 5.41s system 5% cpu 2:29.94 total
build -t optimise-dockerfile .  3.24s user 5.84s system 5% cpu 2:38.57 total
```

Reply:
Hi @mafonso: try building master, making a code change (add ` puts "hello world"` somewhere), then rebuilding - I think that will be significantly slower than this dockerfile as it will re-trigger all the gems to be installed